pallets/post-follows/src/lib.rs

```diff
@@ -0,0 +1,170 @@
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Account is already a post follower.
+        AlreadyPostFollower,
+        /// Account is not a post follower.
+        NotPostFollower,
+        /// Not allowed to follow a hidden post.
+        CannotFollowHiddenPost,
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn post_followers)]
+    pub type PostFollowers<T: Config> =
+        StorageMap<_, Twox64Concat, PostId, Vec<T::AccountId>, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn post_followed_by_account)]
+    pub type PostFollowedByAccount<T: Config> =
+        StorageMap<_, Blake2_128Concat, (T::AccountId, PostId), bool, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn posts_followed_by_account)]
+    pub type PostsFollowedByAccount<T: Config> =
+        StorageMap<_, Twox64Concat, T::AccountId, Vec<PostId>, ValueQuery>;
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        PostFollowed { follower: T::AccountId, post_id: PostId },
+        PostUnfollowed { follower: T::AccountId, post_id: PostId },
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight(<T as Config>::WeightInfo::follow_post())]
+        pub fn follow_post(origin: OriginFor<T>, post_id: PostId) -> DispatchResult {
+            let follower = ensure_signed(origin)?;
+
+            ensure!(
+                !Self::post_followed_by_account((follower.clone(), post_id)),
+                Error::<T>::AlreadyPostFollower
+            );
+
+            let post = Posts::<T>::require_post(post_id)?;
+            ensure!(!post.hidden, Error::<T>::CannotFollowHiddenPost);
+
+            ensure!(
+                T::IsAccountBlocked::is_allowed_account(follower.clone(), post.id),
+                ModerationError::AccountIsBlocked
+            );
+
+            Self::add_post_follower(follower, post_id);
+
+            Ok(())
+        }
+
+        #[pallet::call_index(1)]
+        #[pallet::weight(<T as Config>::WeightInfo::unfollow_post())]
+        pub fn unfollow_post(origin: OriginFor<T>, post_id: PostId) -> DispatchResult {
+            let follower = ensure_signed(origin)?;
+
+            Posts::<T>::ensure_post_exists(post_id)?;
+
+            ensure!(
+                Self::post_followed_by_account((follower.clone(), post_id)),
+                Error::<T>::NotPostFollower
+            );
+
+            Self::remove_post_follower(follower, post_id)
+        }
+
+        #[pallet::call_index(2)]
+        #[pallet::weight((
+            Weight::from_ref_time(100_000) + T::DbWeight::get().reads_writes(3, 4),
+            DispatchClass::Operational,
+            Pays::Yes,
+        ))]
+        pub fn force_follow_post(
```

Comment:
We need to get rid of this `force_***` extrinsic. And plus, we planned to get rid of most of the other `force_***` extrinsics that we used to perform migration from solochain to Kusama parachain.